### PR TITLE
Simplify PKCS#12 PBE algorithm.

### PR DESCRIPTION
**Pull request by snhenson**
Simplify PKCS#12 PBE algorithm by working out Ij + B + 1 directly
instead of using BIGNUMs.

### CODE REVIEW

**Comment by kroeckx** [comment]
Is there a reason to have this in 1.1.0? It doesn't look like a bugfix.

**Comment by snhenson** [comment]
OK, removed 1.1.0

File: crypto/pkcs12/p12_key.c [hunk 1]
```diff
@@ -114,10 +113,7 @@ int PKCS12_key_gen_uni(unsigned char *pass, int passlen, unsigned char *salt,
         Plen = 0;
     Ilen = Slen + Plen;
```
Comment (snhenson) [inline crypto/pkcs12/p12_key.c]: Well the values normally come from trusted sources (PKCS#12 files, encrypted private keys etc) but yes it does make sense to sanity check them. Though that would apply to 1.1.0, 1.0.2 too.

Comment (levitte) [inline crypto/pkcs12/p12_key.c]: So that'd be a fix for a different PR, I take it...

**Comment by davidben** [comment]
By the way, I actually revised this function a while ago, partly to drop the `BIGNUM`s as in this PR, and partly to just tidy it up (variable names, `size_t`, overflow checks, unnecessary allocation of `D` and `Ai`, etc). It's in BoringSSL style rather than OpenSSL style and `size_t` would change the public API, but you all are welcome to treat that function (`pkcs12_key_gen_raw`) as a contribution if you'd like parts of it.

https://boringssl.googlesource.com/boringssl/+/4c0e6c64b6866926f96576bc8e8ea8849f018159/crypto/pkcs8/pkcs8.c#108

**Comment by levitte** [comment]
This is exercised with `80-test_pkcs12.t`, right?

**Comment by richsalz** [comment]
ping @snhenson to merge to master :)
